manifests/repo.pp

```diff
@@ -1,22 +1,43 @@
-# @summary Configure the foreman repo
-# @api private
+# Configure the foreman repo
+#
+# @param repo
+#   The repository version to manage. This can be a specific version or nightly
+#
+# @param configure_scl_repo
+#   If disabled the SCL repo will not be configured on Red Hat clone systems.
+#   (Currently only installs repos for CentOS and Scientific)
+#
+# @param gpgcheck
+#   Turn on/off gpg check in repo files (effective only on RedHat family systems)
+#
+# @param scl_repo_ensure
+#   The ensure to set on the SCL repo package
+#
 class foreman::repo(
-  $repo                = $foreman::repo,
-  $gpgcheck            = $foreman::gpgcheck,
-  $configure_epel_repo = $foreman::configure_epel_repo,
-  $configure_scl_repo  = $foreman::configure_scl_repo,
+  Optional[Variant[Enum['nightly'], Pattern['^\d+\.\d+$']]] $repo = undef,
+  Boolean $gpgcheck = true,
+  Boolean $configure_scl_repo = $facts['os']['family'] == 'RedHat' and $facts['os']['major'] == '7',
+  String $scl_repo_ensure = 'installed',
 ) {
   if $repo {
     foreman::repos { 'foreman':
       repo     => $repo,
       gpgcheck => $gpgcheck,
-      before   => Class['foreman::repos::extra'],
+      before   => Anchor['foreman::repo'],
+    }
+
+    if $configure_scl_repo {
+      Foreman::Repos['foreman'] -> Package['foreman-release-scl']
     }
   }
 
-  class { 'foreman::repos::extra':
-    configure_epel_repo => $configure_epel_repo,
-    configure_scl_repo  => $configure_scl_repo,
+  if $configure_scl_repo {
+    package {'foreman-release-scl':
```

Comment:
Was the intent to move this to centos-release-scl-rh instead?

Reply:
No. I first want to merge it this way and rely on it. I want to drop `foreman-release-scl` in a separate PR but with this merged, I can at least start to clean up acceptance tests in puppet-foreman_proxy which will help with EL8.